wolfcrypt/src/asn.c

```diff
@@ -5612,8 +5642,8 @@ static int DecodePolicyOID(char *out, word32 outSz, byte *in, word32 inSz)
             return ASN_PARSE_E;
         }
 
-        /* Validate total length (2 is the CERT_POLICY_OID+SEQ) */
-        if ((total_length + 2) != sz) {
+        /* Validate total length */
+        if (total_length > (sz - (int)idx)) {
```

Comment:
This total_length check was broken. For my test case the total_length + 2 = 226 while sz 227, which incorrectly failed.